checkov/common/checks_infra/checks_parser.py

```diff
@@ -246,7 +247,11 @@ def _parse_raw_check(self, raw_check: Dict[str, Any], resources_types: Optional[
                     or (isinstance(resource_type, str) and resource_type.lower() == "all")
                     or (isinstance(resource_type, list) and resource_type[0].lower() == "all")
             ):
-                check.resource_types = resources_types or []
+                if bool(convert_str_to_bool(os.getenv('CKV_SUPPORT_ALL_RESOURCE_TYPE', False))):
```

Comment:
when adding a new env_var in checkov, you should add it in `checkov/common/util/env_vars_config.py` and use it from there, allows better handling of those variables throughout the code